managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/MetaStoreImplZookeeper.java

```diff
@@ -330,60 +316,29 @@ public void removeManagedLedger(String ledgerName, MetaStoreCallback<Void> callb
 
     private ManagedLedgerInfo parseManagedLedgerInfo(byte[] data)
             throws ParseException, InvalidProtocolBufferException {
-        if (protobufFormat == ZNodeProtobufFormat.Text) {
-            // First try text format, then fallback to binary
-            try {
-                return parseManagedLedgerInfoFromText(data);
-            } catch (ParseException e) {
-                return parseManagedLedgerInfoFromBinary(data);
-            }
-        } else {
-            // First try binary format, then fallback to text
-            try {
-                return parseManagedLedgerInfoFromBinary(data);
-            } catch (InvalidProtocolBufferException e) {
-                return parseManagedLedgerInfoFromText(data);
-            }
+        // First try binary format, then fallback to text
+        try {
+            return ManagedLedgerInfo.parseFrom(data);
+        } catch (InvalidProtocolBufferException e) {
+            // Fallback to parsing protobuf text format
+            ManagedLedgerInfo.Builder builder = ManagedLedgerInfo.newBuilder();
+            TextFormat.merge(new String(data, Encoding), builder);
```

Comment:
I didn't notice that the method is throwing exception. I was thinking that we are not handling the exception when it fails here.